telegram/bot.py

```diff
@@ -1526,15 +1527,20 @@ def answer_inline_query(self,
                             switch_pm_text=None,
                             switch_pm_parameter=None,
                             timeout=None,
+                            current_offset=None,
                             **kwargs):
         """
         Use this method to send answers to an inline query. No more than 50 results per query are
         allowed.
 
         Args:
             inline_query_id (:obj:`str`): Unique identifier for the answered query.
-            results (List[:class:`telegram.InlineQueryResult`)]: A list of results for the inline
-                query.
+            results (List[:class:`telegram.InlineQueryResult`] | Callable): A list of results for
```

Comment:
E.g. for callbacks we also just state `Callable` and state the signature explicitly in the docstring (see [Handler](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.handler.html)). Personally, I don't find `Callable[[int], Optional[List[:class:`telegram.InlineResult`]]]` overly readible and newbies will not understand it.

> What happens when user returns empty list?

»or :obj:`None`, if there are no more results.« Is that ambiguous

Reply:
There's also `Protocol`:

```
class ResultsGenerator(typing.Protocol):
    def __call__(self, current_index: int) -> typing.List[telegram.InlineResult]:
        ...
```

But would be quite revolutionary  to add that, and even more verbose ;) So just FYI

Reply:
> I don't find Callable[[int], Optional[List[:class:telegram.InlineResult]]] overly readible

Yea, generally that syntax is awful. Takes a bit of getting used to

Reply:
> »or :obj:None, if there are no more results.« Is that ambiguous

Maybe something along the lines of "or an expression that evaluates to False" (bad wording, I know, but you get the gist)

Reply:
> But would be quite revolutionary to add that, and even more verbose ;) So just FYI

good to know :)

> Maybe something along the lines of "or an expression that evaluates to False" (bad wording, I know, but you get the gist)

I'd rather leave that as implementation detail. Documenting it might lead to problems, if the implementation needs to be changed and `None` is easier to graps for newbies